### PR TITLE
177326490 set up email templates

### PR DESCRIPTION
**Pull request by tommywei110**
Gemfile: added panoramic gem
model/email_template.rb: model to store views
mailer/application_mailer.rb: let the controller know to use db before looking at filesystem
fixtures/email_templates.yml: the problem here is i can't seem to put erb things. So might be non-trivial to set up comprehensive tests
seeds.rb: it only destroys admin teachers from the database before seeding, but all email templates are destroyed before seeding.

**Note**: The feature step `Given I seed data` should be placed carefully as it will destroy admin teachers created by Cucumber. As for Rspec, `Rails.application.load_seed` can destroy admin teachers created by fixture if it is ran after `fixtures :all`. At the moment, the seed data must be loaded if the mailer is involved.

### CODE REVIEW

File: db/seed_data.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+module SeedData
+  def self.emails
+    [
+      id: 1,
+      body: "Hi <%= @teacher.first_name %>, Thanks for your feedback. Works",
```
Comment (tommywei110) [inline db/seed_data.rb]: https://www.rubyguides.com/2018/11/ruby-heredoc/

File: spec/teacher_mailer_spec.rb [hunk 2]
```diff
@@ -1,15 +1,15 @@
 require 'rails_helper'
 
 describe TeacherMailer do
-	fixtures :teachers
-	it 'Sends Welcome Email' do
+	fixtures :all
```
Comment (longhtong) [inline spec/teacher_mailer_spec.rb:4]: https://stackoverflow.com/questions/6004057/with-rspec-how-to-seed-the-database-on-load/19930700#19930700

File: app/helpers/email_templates_helper.rb [hunk 3]
```diff
@@ -0,0 +1,2 @@
+module EmailTemplatesHelper
+end
```
Comment (tommywei110) [inline app/helpers/email_templates_helper.rb]: Do we need this?

File: app/mailers/teacher_mailer.rb [hunk 4]
```diff
@@ -1,4 +1,5 @@
 # frozen_string_literal: true
+require 'liquid-rails'
```
Comment (tommywei110) [inline app/mailers/teacher_mailer.rb]: I think this is redundant.

File: app/views/email_templates/delete.html.erb [hunk 5]
```diff
@@ -0,0 +1,2 @@
+<h1>EmailTemplates#delete</h1>
+<p>Find me in app/views/email_templates/delete.html.erb</p>
```
Comment (tommywei110) [inline app/views/email_templates/delete.html.erb]: Disable this if we don't intend to delete.

File: config/routes.rb [hunk 6]
```diff
@@ -8,7 +8,7 @@
   resources :teachers # Note we're not actually using all of these...
   resources :admins   # Note we're not actually using all of these...
   resources :schools, only: [:new, :create, :index]
-
+  resources :email_templates, only: [:index, :update, :edit, :delete]
```
Comment (tommywei110) [inline config/routes.rb]: consider delete 'delete'

**Review by tommywei110 — COMMENTED** [review]
Small things

File: app/helpers/email_templates_helper.rb (hunk 3 above)
Comment (longhtong) [reply app/helpers/email_templates_helper.rb]: Probably not

File: config/routes.rb (hunk 6 above)
Comment (longhtong) [reply config/routes.rb]: Will do

File: spec/fixtures/teachers.yml [hunk 7]
```diff
@@ -8,8 +8,8 @@ ye:
   email: 'ye@berkeley.edu'
```
Comment (nmecklenburg) [inline spec/fixtures/teachers.yml:38]: What is motivating your use of fixtures here? Was this by Michael's guidance?

In the PR description, you mention "fixtures/email_templates.yml: the problem here is i can't seem to put erb things. So might be non-trivial to set up comprehensive tests" -- were you trying to use erb-like syntax to dynamically set values and vary the data to be tested? Fixtures wouldn't be a good match then

Comment (nmecklenburg) [reply spec/fixtures/teachers.yml:38]: Discussed in person: fixtures were already used in the legacy app, so they were extended upon. Recommendation given to switch to something like factories for best practices, but not a top priority for now.

File: app/views/email_templates/update.html.erb [hunk 8]
```diff
@@ -0,0 +1,2 @@
+<h1>EmailTemplates#update</h1>
+<p>Find me in app/views/email_templates/update.html.erb</p>
```
Comment (christopher-hou) [inline app/views/email_templates/update.html.erb]: Not entirely sure what this is, can someone explain?

**Review by christopher-hou — APPROVED** [review]
Looks excellent, testing looks good, good work!

File: app/views/email_templates/update.html.erb (hunk 8 above)
Comment (longhtong) [reply app/views/email_templates/update.html.erb]: It's an auto-generated view by Rails. This should have been deleted. Thanks for catching it!

File: app/assets/javascripts/email_templates.js [hunk 9]
```diff
@@ -0,0 +1,3 @@
+// Place all the behaviors and hooks related to the matching controller here.
+// All this logic will automatically be available in application.js.
+//= require tinymce
```
Comment (cycomachead) [inline app/assets/javascripts/email_templates.js:3]: It might be good to move this to webpack in the future, but if this works, then that's good for now.

**Review by cycomachead — APPROVED** [review]
🎉 This looks good!

**Review by justinphyo — APPROVED** [review]

**Pull request merged**